business/mesh.go

```diff
@@ -133,7 +133,16 @@ func (in *MeshService) discoverKiali(ctx context.Context, clusterName string, r
 	for _, d := range services.Items {
 		kiali := convertKialiServiceToInstance(&d)
 		// If URL is already populated (because of an annotation), trust that because it's user configuration.
-		// Only guess ourselves on our own cluster.
+		// If empty, check Kiali URL configured per cluster name.
+		if kiali.Url == "" {
+			for _, cfgurl := range config.Get().KialiFeatureFlags.Clustering.KialiURLs {
+				if cfgurl.ClusterName == clusterName {
+					kiali.Url = cfgurl.URL
+					break
+				}
```

Comment:
It seems like having `KialiURLs` be an array enables there to be multiple Kiali instances with distinct urls on a single cluster but when looping over them we are just choosing the first URL. How would we even associated the correct URL with the correct kiali instance? Note that `services` here all comes from a single cluster but `var instances []kubernetes.KialiInstance` is an array so presumably there could be multiple kiali instances on a single cluster.

Reply:
I don't see why we need to support multiple kiali instances, but if we're keeping the response the same then it seems like you'll need to add service name/namespace to the `KialiURL` struct in order to associate the correct url with the correct service.

Reply:
@hhovsepy sorry for all the noise. We want to provide the URL config option when the user _doesn't_ have access to the other clusters but still wants a link to an external kiali. I think we need to create the `KialiInstance` from the config URL regardless of whether or not the `GetServices` call fails.